turbinia/config/__init__.py

```diff
@@ -187,3 +192,159 @@ def ValidateAndSetConfig(_config):
       setattr(sys.modules[__name__], var, None)
     else:
       setattr(sys.modules[__name__], var, getattr(_config, var))
+
+class TurbiniaRecipe(object):
+  """ Base class for Turbinia recipes
+
+  Attributes:
+      recipe_file (str): name of the recipe file to be loaded.
+      jobs_whitelist (list): A whitelist for Jobs that will be allowed to run.
+      jobs_blacklist (list): A blacklist for Jobs that will not be allowed to run.
+      filter_patterns_file (str): Path to a file containing newline separated string patterns with which to filter text
+          based evidence.
+      task_recipes (dict): Object containing a task specific recipe for each of the tasks invoked in the Turbinia 
+          recipe.
+"""
+
+  def __init__(self, recipe_file, filter_patterns_files=[]):
+    self.recipe_file = recipe_file
+    self.filter_patterns_files = filter_patterns_files if filter_patterns_files else []
+
+    self.name = ""
+    self.jobs_whitelist = []
+    self.jobs_blacklist = []
+    self.filter_patterns = []    
+    self.task_recipes = {}
+ 
+  def load(self):
+    # Assumes a main turbinia config is available, as recipe and pattern file directories must be present.
+    if not CONFIG:
+      LoadConfig()
+    with open(self.recipe_file, 'r') as r_file:
+      recipe_contents = r_file.read()
+    recipe_dict = load(recipe_contents, Loader=Loader)
```

Comment:
put in an exception to catch yaml.parser.ParserError, which is the exception that fires when a malformed yaml file is passed. Should we consider putting in a safeguard to ensure nothing but job_*lists, pattern_file, or existing task names are passed as keys? This would prevent turbinia from ignoring a recipe on which that someone made a typo.

Reply:
Yeah, I think that would be nice to have, but not a super high priority.